Simplenote/SPAppDelegate.m

```diff
@@ -683,6 +704,17 @@ - (void)bucketDidFinishIndexing:(SPBucket *)bucket
 {
     if ([bucket.name isEqualToString:@"Note"]) {
         [_noteListViewController setWaitingForIndex:NO];
+        
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
+            NSArray *deleted = [bucket objectsForPredicate:[NSPredicate predicateWithFormat:@"deleted == YES"]];
+            if (deleted.count > 0) {
+                NSArray *deletedIDs = [deleted valueForKey:@"simperiumKey"];
+                [[CSSearchableIndex defaultSearchableIndex] deleteSearchableItemsWithIdentifiers:deletedIDs completionHandler:nil];
+            }
+            
+            NSArray *notes = [bucket objectsForPredicate:[NSPredicate predicateWithFormat:@"deleted == NO"]];
+            [[CSSearchableIndex defaultSearchableIndex] indexSearchableNotes:notes];
+        });
```

Comment:
Although `didFinishIndexing` is a suitable place to do this, there are few problems with this snippet:

- `SPBucket` is not thread safe. `objectsForPredicate` runs agains the Main Thread MOC, which would potentially cause issues when / if called from a BG thread
- The `deleted` flag is just a temporary indicator, which (AFAIK) will be gone after the context effectively saves. This could cause erratic behavior, due to the Core Data architecture (BG writer saving asynchronously). I _suspect_ inspecting the `deleted` property of an object that's gone, but still registered in the MOC, may break the app.

How about if, instead, we move the `indexSearchableNote` / `deleteSearchableItems` calls to this SPBucket delegate method?

```
bucket:(SPBucket *)bucket didChangeObjectForKey:(NSString *)key forChangeType:(SPBucketChangeType)change memberNames:(NSArray *)memberNames
```